File: python/allergies/allergies.py
class Allergies:
  def __init__(self, allergy_index):
    self.allergy_index = allergy_index
    self.allergies = {
      'eggs': 1,
      'peanuts': 2,
      'shellfish': 4,
      'strawberries': 8,
      'tomatoes': 16,
      'chocolate': 32,
      'pollen': 64,
      'cats': 128
    }

  @property
  def lst(self):
    return [allergy for allergy in self.allergies if self.is_allergic_to(allergy)]

  def is_allergic_to(self, item):
    return self.allergy_index & self.allergies[item]

File: csharp/allergies/Allergies.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Allergies
{
    private readonly int allergyIndex;

    private enum Allergen
    {
        Eggs = 1,
        Peanuts = 2,
        Shellfish = 4,
        Strawberries = 8,
        Tomatoes = 16,
        Chocolate = 32,
        Pollen = 64,
        Cats = 128 
    }

    public Allergies(int allergyIndex)
    {
        this.allergyIndex = allergyIndex;
    }

    public bool AllergicTo(string item)
    {
        Allergen allergen;
        return Enum.TryParse(item, true, out allergen) && IsAllergicTo(allergen);
    }

    private bool IsAllergicTo(Allergen allergen)
    {
        return ((int) allergen & allergyIndex) != 0;
    }

    public List<string> List()
    {
        return 
            Enum.GetValues(typeof(Allergen))
                .Cast<Allergen>()
                .Where(IsAllergicTo)
                .Select(x => x.ToString().ToLower())
                .ToList();
    }
}